src/index.js

```diff
@@ -53,6 +53,13 @@ class FloodSub extends EventEmitter {
      */
     this.subscriptions = new Set()
 
+    /**
+     * Cache for the last message on seen subscriptions
+     * @type {Map<string, Buffer>}
+     * @private
+     */
+    this._lvc = new Map()
```

Comment:
I would suggest using LRU cache, caches like that bitten as so many times in go-ipfs as things scaled up.